homeassistant/components/azure_data_explorer/__init__.py

```diff
@@ -0,0 +1,235 @@
+    """Activate ADX component from yaml.
+
+    Adds an empty filter to hass data.
+    Tries to get a filter from yaml, if present set to hass data.
+    If config is empty after getting the filter, return, otherwise emit
+    deprecated warning and pass the rest to the config flow.
+    """
+
+    hass.data.setdefault(DOMAIN, {DATA_FILTER: FILTER_SCHEMA({})})
+    if DOMAIN not in yaml_config:
+        return True
+    hass.data[DOMAIN][DATA_FILTER] = yaml_config[DOMAIN][CONF_FILTER]
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Do the setup based on the config entry and the filter from yaml."""
+    hass.data.setdefault(DOMAIN, {DATA_FILTER: FILTER_SCHEMA({})})
+    adx = AzureDataExplorer(hass, entry)
+    try:
+        await adx.test_connection()
+    except KustoServiceError as exp:
+        _LOGGER.error(exp)
+        raise IntegrationError(
+            "Could not find Azure Data Explorer database or table"
+        ) from exp
+    except KustoAuthenticationError as exp:
+        _LOGGER.error(exp)
+        raise ConfigEntryNotReady(
+            "Could not authenticate to Azure Data Explorer"
+        ) from exp
+    except Exception as exp:  # pylint: disable=broad-except
+        _LOGGER.error(exp)
+        raise ConfigEntryNotReady("Could not connect to Azure Data Explorer") from exp
+    hass.data[DOMAIN][DATA_HUB] = adx
+    entry.async_on_unload(entry.add_update_listener(async_update_listener))
+    await adx.async_start()
+    return True
+
+
+async def async_update_listener(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Update listener for options."""
+    hass.data[DOMAIN][DATA_HUB].update_options(entry.options)
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    adx = hass.data[DOMAIN].pop(DATA_HUB)
+    await adx.async_stop()
+    return True
+
+
+class AzureDataExplorer:
+    """A event handler class for Azure Data Explorer."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        entry: ConfigEntry,
+    ) -> None:
+        """Initialize the listener."""
+
+        self.hass = hass
+        self._entry = entry
+        self._entities_filter = hass.data[DOMAIN][DATA_FILTER]
+
+        self._client = AzureDataExplorerClient(
+            clusteringesturi=self._entry.data["clusteringesturi"],
+            database=self._entry.data["database"],
+            table=self._entry.data["table"],
+            client_id=self._entry.data["client_id"],
+            client_secret=self._entry.data["client_secret"],
+            authority_id=self._entry.data["authority_id"],
+            use_free_cluster=self._entry.data["use_free_cluster"],
+        )
+
+        self._send_interval = self._entry.options[CONF_SEND_INTERVAL]
+        self._max_delay = self._entry.options.get(CONF_MAX_DELAY, DEFAULT_MAX_DELAY)
```

Comment:
Does this options still exist? or should it just set that value or skip this variable entirely and just use the DEFAULT?